plugins/edge/factory.go

```diff
@@ -19,13 +20,15 @@ type PluginFactory struct {
 	ctx    context.Context
 	cfg    *topaz.Config
 	logger *zerolog.Logger
+	app *app.Topaz
```

Comment:
If we change SetServiceStatus to be a func instead of a method, we do not need any changes to the test engine.